member-service/src/main/java/com/hedvig/memberservice/aggregates/MemberAggregate.kt

```diff
@@ -0,0 +1,588 @@
+                    cmd.apartmentNo,
+                    floor))
+        }
+        if (cmd.phoneNumber != null
+            && member.phoneNumber != cmd.phoneNumber) {
+            apply(PhoneNumberUpdatedEvent(id, cmd.phoneNumber))
+        }
+        if (cmd.birthDate != null
+            && member.birthDate != cmd.birthDate) {
+            apply(BirthDateUpdatedEvent(id, cmd.birthDate))
+        }
+    }
+
+    @CommandHandler
+    fun bankIdSignHandler(cmd: SwedishBankIdSignCommand) {
+        if (cmd.personalNumber != null
+            && member.ssn != cmd.personalNumber) {
+            apply(SSNUpdatedEvent(id, cmd.personalNumber, SSNUpdatedEvent.Nationality.SWEDEN))
+        }
+        getBirthdateFromSwedishSsnAndApplyEvent(cmd.personalNumber, null)
+        apply(NewCashbackSelectedEvent(id, cashbackService.getDefaultId(id).toString()))
+        apply(
+            MemberSignedEvent(
+                id, cmd.referenceId, cmd.signature, cmd.oscpResponse,
+                cmd.personalNumber))
+        if (trackingId == null) {
+            generateTrackingId()
+        }
+    }
+
+    @CommandHandler
+    fun ZignSecSignHandler(cmd: ZignSecSignCommand) {
+        if (!isValidJSON(cmd.provideJsonResponse)) throw RuntimeException("Invalid json from provider")
+        apply(NewCashbackSelectedEvent(id, cashbackService.getDefaultId(id).toString()))
+        val zignSecAuthMarket = cmd.zignSecAuthMarket
+        if (zignSecAuthMarket === ZignSecAuthenticationMarket.NORWAY) {
+            if (cmd.personalNumber != null
+                && member.ssn != cmd.personalNumber) {
+                apply(NorwegianSSNUpdatedEvent(id, cmd.personalNumber))
+            }
+            apply(
+                NorwegianMemberSignedEvent(
+                    id, cmd.personalNumber, cmd.provideJsonResponse, cmd.referenceId))
+            return
+        } else if (zignSecAuthMarket === ZignSecAuthenticationMarket.DENMARK) {
+            if (cmd.personalNumber != null
+                && member.ssn != cmd.personalNumber) {
+                apply(DanishSSNUpdatedEvent(id, cmd.personalNumber))
+            }
+            apply(
+                DanishMemberSignedEvent(
+                    id, cmd.personalNumber, cmd.provideJsonResponse, cmd.referenceId))
+            return
+        }
+        throw RuntimeException("ZignSec authentication market: " + zignSecAuthMarket.name + " is not implemented!")
+    }
+
+    fun isValidJSON(json: String?): Boolean {
+        return try {
+            objectMapper.readTree(json)
+            true
+        } catch (e: IOException) {
+            logger.error("Failed to validate json", e)
+            false
+        }
+    }
+
+    @CommandHandler
+    fun on(signMemberFromUnderwriterCommand: SignMemberFromUnderwriterCommand) {
+        apply(MemberSignedWithoutBankId(signMemberFromUnderwriterCommand.id, signMemberFromUnderwriterCommand.ssn))
+    }
+
+    @CommandHandler
+    fun on(cmd: MemberSimpleSignedCommand) {
```

Comment:
`handle` for commands, `on` for events
```suggestion
    fun handle(cmd: MemberSimpleSignedCommand) {
```